api/app/Models/PoolCandidate.php

```diff
@@ -730,11 +730,16 @@ public function isDraft()
     /**
      * Scope the query to PoolCandidate's the current user can view
      */
-    public function scopeAuthorizedToView(Builder $query): void
+    public function scopeAuthorizedToView(Builder $query, ?array $args = null): void
     {
         /** @var \App\Models\User */
         $user = Auth::user();
 
+        $userId = $args['userId'] ?? null;
+        if ($userId) {
+            $user = User::find($userId);
```

Comment:
Maybe findOrFail?  Is there a reason we would want to proceed if the given $userId doesn't exists?  Also, isn't `?? null` redundant?

Reply:
Same comments for api/app/Models/User.php L1045.

Reply:
Oops, I was thinking of `array_key_exists`. This should do the same thing: [adjust userId access in authorized scopes](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11300/commits/4bcb96c1a12d4ccbb6627b23dbf8a24d994d2623)